language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/MakeTypePublicCodeAction.java

```diff
@@ -0,0 +1,128 @@
+
+import io.ballerina.compiler.api.ModuleID;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.syntax.tree.ClassDefinitionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.TypeDefinitionNode;
+import io.ballerina.projects.Project;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.langserver.common.constants.CommandConstants;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.commons.CodeActionContext;
+import org.ballerinalang.langserver.commons.codeaction.spi.DiagBasedPositionDetails;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Code Action to make type public.
+ *
+ * @since 2201.0.3
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.codeaction.spi.LSCodeActionProvider")
+public class MakeTypePublicCodeAction extends AbstractCodeActionProvider {
+    public static final String NAME = "Make Type Public";
+    public static final String DIAGNOSTIC_CODE = "BCE2038";
+
+    @Override
+    public List<CodeAction> getDiagBasedCodeActions(Diagnostic diagnostic,
+                                                    DiagBasedPositionDetails positionDetails,
+                                                    CodeActionContext context) {
+        if (!DIAGNOSTIC_CODE.contains(diagnostic.diagnosticInfo().code()) || context.currentSyntaxTree().isEmpty()
+                || context.currentSemanticModel().isEmpty()) {
+            return Collections.emptyList();
+        }
+        Range diagnosticRange = new Range(CommonUtil.toPosition(diagnostic.location().lineRange().startLine()),
+                CommonUtil.toPosition(diagnostic.location().lineRange().endLine()));
+        NonTerminalNode nonTerminalNode = CommonUtil.findNode(diagnosticRange, context.currentSyntaxTree().get());
+        Optional<Symbol> symbol = context.currentSemanticModel().get().symbol(nonTerminalNode);
+        if (symbol.isEmpty() || symbol.get().getModule().isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        ModuleID moduleID = symbol.get().getModule().get().id();
+        String orgName = moduleID.orgName();
+        String moduleName = moduleID.moduleName();
+        Optional<Project> project = context.workspace().project(context.filePath());
+        if (project.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Optional<Path> filePath = CommonUtil.getFilePathForDependency(orgName, moduleName, project.get(),
```

Comment:
We can reuse the logic:
```java
ModuleID moduleID = symbol.get().getModule().get().id();
        String orgName = moduleID.orgName();
        String moduleName = moduleID.moduleName();
        Optional<Project> project = context.workspace().project(context.filePath());
        if (project.isEmpty()) {
            return Collections.emptyList();
        }
        Optional<Path> filePath = CommonUtil.getFilePathForDependency(orgName, moduleName, project.get(),
```
Shall we create a new utility method in `CommonUtil` named `getFilePathForSymbol(Symbol, Project, DocumentServiceContext)`.

Reply:
Done